sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.institution.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { Controller, Get, Param, ParseIntPipe } from "@nestjs/common";
+import BaseController from "../BaseController";
+import {
+  RestrictionSummaryAPIOutDTO,
+  RestrictionDetailAPIOutDTO,
+} from "./models/restriction.dto";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import {
+  AllowAuthorizedParty,
+  HasStudentDataAccess,
+  IsBCPublicInstitution,
+} from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+import { ApiTags } from "@nestjs/swagger";
+import { RestrictionControllerService } from "./restriction.controller.service";
+
+/**
+ * Controller for Institution Restrictions.
+ * This consists of all Rest APIs for Institution restrictions.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@IsBCPublicInstitution()
+@Controller("restriction")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-restriction`)
+export class RestrictionInstitutionController extends BaseController {
+  constructor(
+    private readonly restrictionControllerService: RestrictionControllerService,
+  ) {
+    super();
+  }
+
+  /**
+   * Get restrictions for a student filtered with restriction notification type other than `No effect` or restriction category other than `Verification`.
+   * @param studentId id of the student to retrieve restrictions.
+   * @returns Student restrictions.
+   */
+  @HasStudentDataAccess("studentId")
+  @Get("student/:studentId")
+  async getStudentRestrictions(
+    @Param("studentId", ParseIntPipe) studentId: number,
+  ): Promise<RestrictionSummaryAPIOutDTO[]> {
+    return this.restrictionControllerService.getStudentRestrictions(studentId, {
+      filterRestrictions: true,
+    });
+  }
+
+  /**
+   * Get the details for view student restriction.
+   * @param studentId id of the student.
+   * @param studentRestrictionId id of the student restriction.
+   * @returns Student restriction detail view.
+   */
+  @HasStudentDataAccess("studentId")
```

Comment:
This decorator can be moved to the class level as every endpoint is using it.

Reply:
but what if there is will be an API endpoint with no `student id`?

Reply:
@ann-aot We assume there won't be such an endpoint for this controller.

Reply:
I don't want to hold the PR, for the PR  I am okay with the change, I was wondering what if the institution needs a list of restrictions or something related to restrictions, which is not related to a particular student. 
Not a blocker of this PR and we can resolve the comment and keep the change we have now.

Reply:
I hear what you are saying, we can see when we get such endpoint.